code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm

```diff
@@ -467,3 +466,454 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
+	. = ..()
+	if(!.)
+		return FALSE
+	if(held_item)
+		return TRUE
+	if(!isitem(A) || isgrabitem(A))
+		if(!silent)
+			A.balloon_alert(owner, "not an item")
+		return FALSE
+	var/obj/item/item_atom = A
+	if(!owner.Adjacent(item_atom))
+		if(!silent)
+			item_atom.balloon_alert(owner, "too far")
+		return FALSE
+	if(item_atom.anchored)
+		if(!silent)
+			item_atom.balloon_alert(owner, "item is anchored")
+		return FALSE
+
+/datum/action/ability/activable/xeno/item_throw/use_ability(atom/A)
+	if(!held_item)
+		playsound(owner, 'sound/voice/alien/pounce2.ogg', 30)
+		var/obj/item/interacted_item = A
+		interacted_item.forceMove(owner)
+		held_item = interacted_item
+		held_appearance = mutable_appearance(interacted_item.icon, interacted_item.icon_state)
+		held_appearance.layer = ABOVE_OBJ_LAYER
+		RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
+		RegisterSignal(owner, COMSIG_ATOM_DIR_CHANGE, PROC_REF(on_owner_turn))
+		RegisterSignal(owner, COMSIG_MOB_DEATH, PROC_REF(drop_item))
+		RegisterSignal(owner, COMSIG_MOB_STAT_CHANGED, PROC_REF(drop_item)) // No need to check for the specifics regarding stat as anything that isn't CONSCIOUS should cause it to drop.
+		RegisterSignal(held_item, COMSIG_QDELETING, PROC_REF(on_item_qdel))
+
+		owner.add_movespeed_modifier(MOVESPEED_ID_OPPRESSOR_ITEM_GRAB, TRUE, 0, NONE, TRUE, 2)
+		on_owner_turn(null, null, owner.dir)
+		succeed_activate()
+		ability_cost = 0 // Throwing will cost nothing to prevent the ability from failing to recast if they happen to have not enough plasma.
+		used_movement_allowance = FALSE
+		return
+	owner.remove_movespeed_modifier(MOVESPEED_ID_OPPRESSOR_ITEM_GRAB)
+	held_item.throwforce += min(held_item.w_class * 15, 90) // Upper limit to prevent any weird weight classes (e.g. above WEIGHT_CLASS_GIGANTIC)
+	RegisterSignal(held_item, COMSIG_MOVABLE_POST_THROW, PROC_REF(on_throw_end))
+	held_item.forceMove(get_turf(owner))
+	// A speed of 5 is required to inflict maximum damage to mobs.
+	held_item.throw_at(A, max(2, 11 - (held_item.w_class * 2)), 5)
+	held_item = null
+	owner.overlays -= held_appearance
+	playsound(xeno_owner, 'sound/effects/throw.ogg', 30, 1)
+	succeed_activate()
+	ability_cost = initial(ability_cost)
+	add_cooldown()
+	UnregisterSignal(xeno_owner, list(COMSIG_MOVABLE_MOVED, COMSIG_ATOM_DIR_CHANGE, COMSIG_MOB_DEATH, COMSIG_MOB_STAT_CHANGED))
+
+/// Reduces throwforce by what it was increased by.
+/datum/action/ability/activable/xeno/item_throw/proc/on_throw_end(datum/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(source, list(COMSIG_MOVABLE_POST_THROW, COMSIG_QDELETING))
+	var/obj/item/item_source = source
+	item_source.throwforce -= min(item_source.w_class * 15, 90)
+
+/// On move, drop the item if they used their movement allowance. Otherwise, use up their allowance.
+/datum/action/ability/activable/xeno/item_throw/proc/on_move()
+	SIGNAL_HANDLER
+	if(!used_movement_allowance)
```

Comment:
This is fine I guess, but what you're really saying is 'has the owner moved' (not how much they've moved etc) so I try and make vars simple and clear. i.e. owner_moved or something